geopandas/array.py

```diff
@@ -1258,7 +1258,35 @@ def _values_for_argsort(self):
         ExtensionArray.argsort
         """
         # Note: this is used in `ExtensionArray.argsort`.
-        raise TypeError("geometries are not orderable")
+        from geopandas.tools.hilbert_curve import _hilbert_distance
+
+        if self.size == 0:
+            # TODO _hilbert_distance fails for empty array
+            return np.array([], dtype="uint32")
+
+        mask_missing = self.isna()
+        mask_empty = self.is_empty
+        mask = mask_missing | mask_empty
+        if mask.any():
+            geoms = self.copy()
+            indices = np.nonzero(~mask)[0]
+            if indices.size:
+                geom = self[indices[0]]
+            else:
+                # for all-empty/NA, just take random geometry
+                geom = shapely.geometry.Point(0, 0)
+
+            geoms[mask] = geom
+        else:
+            geoms = self
+
+        return _hilbert_distance(geoms)
+
+    def argmin(self, skipna: bool = True) -> int:
+        raise TypeError("geometries have no minimum or maximum")
+
+    def argmax(self, skipna: bool = True) -> int:
+        raise TypeError("geometries have no minimum or maximum")
```

Comment:
Another question is whether we want to allow getting the `idxmin`/`idxmax`, and also `min`/`max`, on a geometry column, now it is order-able.  
On the one hand, if you can sort, you have a first/last value, and thus min/max. On the other hand I think one can also argue that it's not because we implement ordering the values deterministically, it also has a notion of a min or max.

Reply:
We sort spatially by _location_ not by a _value_, so I would say that `min` `max` is still not defined. PostGIS does return a geometry for MIN MAX queries but it seems to be based on the string representation of binary geometry.

![Screenshot 2023-02-11 at 22 24 04](https://user-images.githubusercontent.com/36797143/218281748-f139ebd2-1ed7-4a26-bf13-86365af8e735.png)